internal/controller/datadogagent/experimental/image_test.go

```diff
@@ -0,0 +1,236 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package experimental
+
+import (
+	"fmt"
+	"testing"
+
+	apicommon "github.com/DataDog/datadog-operator/api/datadoghq/common"
+	"github.com/DataDog/datadog-operator/api/datadoghq/v2alpha1"
+	"github.com/DataDog/datadog-operator/internal/controller/datadogagent/feature/fake"
+	"github.com/stretchr/testify/assert"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/log/zap"
+)
+
+type imageOverrideTestDef struct {
+	name string
+	tag  string
+}
+
+func (i *imageOverrideTestDef) ImageOverrideJSON(containerName string) string {
+	imageOverrides := imageOverrides{
+		containerName: imageOverride{
+			Name: i.name,
+			Tag:  i.tag,
+		},
+	}
+
+	json, err := imageOverrides.JSON()
```

Comment:
nit - this test basically marshals `imageOverride` struct into json, sets it in annotation and then tests the logic, including unmarshalling. Is the goal of the test to verify unmarshalling as well? if not, why can't we just use `imageOverrides` map?
imo, having constant jsons in setup would be more readable and also test unmarshalling of predefined jsons.

Reply:
The intent is to make sure that we we're passing the intended structure: if `imageOverrides` changes over time, our test doesn't become stale because we're always generating it from the same struct that we deserialize into.

All of that said, I'm not terribly partial to this aspect and if you think it's more readable to generate the JSON directly, I'll make that change.